### DISCUSSION TITLE
How to access domain via codespace command line?

### DISCUSSION THREAD

**Discussion by Flux159** [community/community#5104] [Codespaces] [answered]
Hi, I have a next.js app that is working in a codespace & I was wondering how I could configure redirects properly with the github preview domain. Right now server side redirects are redirecting to http://localhost:<port>, which is not ideal since in a codespace preview, I would want it to redirect to the .githubpreview.dev domain.

Specifically it looks like the urls for the codespace vscode ui is at https://"repo-owner"-"repo"-"some-hash".github.dev/ and a forwarded port is defaulted to https://"repo-owner"-"repo"-"some-hash"-"port".githubpreview.dev/ 

Next.js allows you to [specify a hostname when running in dev mode](https://github.com/vercel/next.js/pull/1017) with a '-H' flag so I was wondering how I could access the preview domain name via command line to pass into my app.

Essentially the question is if there's a command that I can run in a codespace container that will allow me to get what the hash    of a public preview url is. For reference `hostname` just returns something like `codespaces_<different_hash>`.

**Comment by Flux159** [comment, marked as answer]
I was able to get the domain of the codespace by using [jq](https://stedolan.github.io/jq/) and reading from the `/workspaces/.codespaces/shared/environment-variables.json` file.

```bash
jq -r ".CODESPACE_NAME" /workspaces/.codespaces/shared/environment-variables.json
```

This works for my next.js use case since the redirect is happening in a browser. 

I noticed that trying to make requests from python to that domain returns an error - I believe that this is due to an authentication mechanism that codespaces uses on initial loading to the codespaces domain (can see this when opening a port in browser & see some token passed in a url before the redirect to the githubpreview.dev domain). This wasn't so much of a problem since python in the container can just use localhost.

**Comment by artis3n** [comment]
Easier than parsing it out of the json file - It is available in the `CODESPACE_NAME` env var. `echo ${CODESPACE_NAME}`

**Comment by artificial-cassiano** [comment]
If I want to access other (custom) parameters on my url? 
Let's say, I have this URL:
```
https://my-codespace-vwq5a2v35gw4.github.dev/?parameter0=lucas
```

and I want to read `parameter0`. How can I do this?

**Reply by artificial-cassiano** [reply]
I tried accessing `USER_PARAMETERS_URL` (defined at`/workspaces/.codespaces/shared/environment-variables.json`), but seems to be `null` only. Is there documentation on how to populate this env var?

**Comment by jarrodonlo** [comment]
### Using port `8080` as an example...

```bash
# shell example
PORT=8080 # desired port
URL="https://${CODESPACE_NAME}-${PORT}.githubpreview.dev"
# echo ${url}
```

```javascript
// javascript example
const port = 8080 // desired port
const url = `https://${process.env["CODESPACE_NAME"]}-${port}.githubpreview.dev`
// console.log(url)
```
